fw/http.c

```diff
@@ -4591,17 +4591,28 @@ tfw_h2_resp_add_loc_hdrs(TfwHttpResp *resp, const TfwHdrMods *h_mods,
 	unsigned int i;
 	TfwHttpTransIter *mit = &resp->mit;
 	TfwH2TransOp op = cache ? TFW_H2_TRANS_EXPAND : TFW_H2_TRANS_ADD;
+	TfwHttpHdrTbl *ht = resp->h_tbl;
 
 	if (!h_mods)
 		return 0;
 
 	for (i = 0; i < h_mods->sz; ++i) {
 		const TfwHdrModsDesc *desc = &h_mods->hdrs[i];
 		int r;
+		unsigned short hid = desc->hid;
 
 		/* FIXME: this is a temporary WA for GCC12, see #1695 for details */
-		if (test_bit(i, mit->found) || (TFW_STR_CHUNK(desc->hdr, 1) == NULL))
+		if (test_bit(i, mit->found)
```

Comment:
After fixing `resp_hdr_add` for cached responses `test_bit(i, mit->found)` can be deleted. Because it's not used anymore for http2 headers addition.